hyperspace/solana/src/lib.rs

```diff
@@ -0,0 +1,253 @@
+#[derive(Clone)]
+pub struct Client {
+	/// Chain name
+	pub name: String,
+	/// rpc url for cosmos
+	pub rpc_url: String,
+	/// Solana chain Id
+	pub chain_id: String,
+	/// Light client id on counterparty chain
+	pub client_id: Option<ClientId>,
+	/// Connection Id
+	pub connection_id: Option<ConnectionId>,
+	/// Account prefix
+	pub account_prefix: String,
+	pub fee_denom: String,
+	/// The key that signs transactions
+	pub keybase: KeyEntry,
+	/// Maximun transaction size
+	pub max_tx_size: usize,
+	pub commitment_level: CommitmentLevel,
+	pub program_id: Pubkey,
+	pub common_state: CommonClientState,
+}
+
+pub struct ClientConfig {
+	/// Chain name
+	pub name: String,
+	/// rpc url for cosmos
+	pub rpc_url: Url,
+	/// Solana chain Id
+	pub chain_id: String,
+	/// Light client id on counterparty chain
+	pub client_id: Option<ClientId>,
+	/// Connection Id
+	pub connection_id: Option<ConnectionId>,
+	/// Account prefix
+	pub account_prefix: String,
+	/// Fee denom
+	pub fee_denom: String,
+	/// Fee amount
+	pub fee_amount: String,
+	/// Fee amount
+	pub gas_limit: u64,
+	/// Store prefix
+	pub store_prefix: String,
+	/// Maximun transaction size
+	pub max_tx_size: usize,
+	/// All the client states and headers will be wrapped in WASM ones using the WASM code ID.
+	pub wasm_code_id: Option<String>,
+	pub common_state_config: CommonClientConfig,
+}
+
+#[derive(Clone)]
+pub struct KeyEntry {
+	pub public_key: Pubkey,
+	pub private_key: Vec<u8>,
+}
+
+impl Client {}
+
+impl IbcProvider for Client {}
+
+impl KeyProvider for Client {
+	fn account_id(&self) -> Signer {
+		let key_entry = self.keybase;
+		let public_key = key_entry.public_key;
+		Signer::from_str(&public_key.to_string()).unwrap()
+	}
+}
+
+#[async_trait::async_trait]
+impl MisbehaviourHandler for Client {
+	async fn check_for_misbehaviour<C: Chain>(
+		&self,
+		_counterparty: &C,
+		_client_message: AnyClientMessage,
+	) -> Result<(), anyhow::Error> {
+		Ok(())
+	}
+}
+
+#[async_trait::async_trait]
+impl LightClientSync for Client {
+	async fn is_synced<C: Chain>(&self, _counterparty: &C) -> Result<bool, anyhow::Error> {
+		Ok(true)
+	}
+
+	async fn fetch_mandatory_updates<C: Chain>(
+		&self,
+		_counterparty: &C,
+	) -> Result<(Vec<Any>, Vec<IbcEvent>), anyhow::Error> {
+		Ok((vec![], vec![]))
+	}
+}
+
+#[async_trait::async_trait]
+impl Chain for Client {
+	fn name(&self) -> &str {
+		&self.name
+	}
+
+	fn block_max_weight(&self) -> u64 {
+		self.max_tx_size as u64
+	}
+
+	async fn estimate_weight(&self, msg: Vec<Any>) -> Result<u64, Self::Error> {
+		todo!()
+	}
+
+	async fn finality_notifications(
+		&self,
+	) -> Result<
+		Pin<Box<dyn Stream<Item = <Self as IbcProvider>::FinalityEvent> + Send + Sync>>,
+		Error,
+	> {
+		todo!()
+	}
+
+	async fn submit(&self, messages: Vec<Any>) -> Result<Self::TransactionId, Error> {
+		let cluster = Cluster::from_str(&self.rpc_url).unwrap();
+		let authority = Rc::new(Keypair::from_bytes(&self.keybase.private_key).unwrap());
```

Comment:
Why RC?

Reply:
The docs used `Rc` but we can remove it since we only need immutable reference of the keypair.